src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,119 @@
+import * as clc from "cli-color";
+
+import * as api from "./api";
+import * as FirebaseError from "./error";
+import * as logger from "./logger";
+import { OperationPollerOptions, pollOperation } from "./operation-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum ParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export async function createFirebaseProject(
```

Comment:
Would prefer signature of:

    createFirebaseProject(projectId: string, options: {projectDisplayName?: string, parentResource?: ParentResource} = {})

AFAIK display name isn't required, right? It should just default to the `projectId`.

Reply:
No it isn't required. And sure I'm on board with your suggestion.

Reply:
Updated the `displayName` to default to the `projectId`

Reply:
Oops, can we do just `displayName` instead of `projectDisplayName`?

Reply:
Gotcha! I named it `displayName`